xbmc/addons/InputStream.cpp

```diff
@@ -119,6 +120,7 @@ bool CInputStream::Open(CFileItem &fileitem)
     props.m_nCountInfoValues++;
   }
   props.m_strURL = fileitem.GetPath().c_str();
+  props.m_libFolder = URIUtils::GetDirectory(m_parentLib).c_str();
```

Comment:
this string (and others above it) need a buffer to hold their value that lasts until Open() is finished. As it is, a temporary string is allocated, but is deallocated after this statement concludes. This works because the memory remains unchanged after the deallocation, but OFC this isn't reliable behaviour